grpcutil/naming/dns_resolver.go

```diff
@@ -0,0 +1,277 @@
+package naming
+
+// Copied from https://github.com/grpc/grpc-go/tree/v1.29.x/naming.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"google.golang.org/grpc/grpclog"
```

Comment:
If this becomes part of dskit, we should switch to go-kit logging.

Reply:
@pstibrany done.